dev/buildtool/git_support.py

```diff
@@ -815,20 +819,20 @@ def find_newest_tag_and_common_commit_from_id(
 
     return start_tag, start_commit
 
-
   def query_local_repository_commits_to_existing_tag_from_id(
-      self, git_dir, commit_id, commit_tags):
+      self, git_dir, commit_id, commit_tags, base_commit_id=None):
     """Returns the list of commit messages to the local repository."""
     # pylint: disable=invalid-name
 
     tag, found_commit = self.find_newest_tag_and_common_commit_from_id(
```

Comment:
Is this something expensive we can avoid if `base_commit_id` is specified?

Reply:
I wanted to keep it simple and not break anything. The problem is that this method is also used to get the tag to use. The tag needs to analyze the absolute changelog even if the caller is only interested in the messages for the relative changelog.

This is offline processing and changelogs are probably small so performance isnt a concern here. It would be nice if things where different but isnt obvious how to restructure it and not worth the effort to pursue further at this time. So yes, this is messy and adds some complexity being overloaded. It might not add runtime cost because the work might be necessary regardless. It would still be desirable to reduce the maintainence / complexity costs but payoff doesnt seem worth it right now.

Reply:
SGTM.